### DISCUSSION TITLE
Security Mutation Operator

### DISCUSSION THREAD

**Discussion by arasyid10** [infection/infection#1914] [Q&A]
Does the infection have an operator to test injection or security domain? Is it possible to add an operator to the infection? How can I add operators to the infection?

**Comment by theofidry** [comment]
You currently cannot create custom operators outside of infection itself, i.e. the way to add one is to contribute one to Infection.

To test injection or security domain however, I am not sure infection is the right tool for it. Indeed, if you want to test injections, what you will act on is the user input, which is something an external actor has control over, rather than source code which no user (should) have access to.

To this there is a few other more suited tools:

- [Psalm taint analysis](https://psalm.dev/articles/detect-security-vulnerabilities-with-psalm)
- [A fuzzing tool](https://github.com/nikic/PHP-Fuzzer)
- And I think you could find a couple more in https://github.com/guardrailsio/awesome-php-security
